src/views/W3NRemove/W3NRemove.tsx

```diff
@@ -0,0 +1,207 @@
+import { useCallback } from 'react';
+import { Link, generatePath } from 'react-router-dom';
+import { browser } from 'webextension-polyfill-ts';
+import { Deposit, IDidIdentifier } from '@kiltprotocol/types';
+
+import { FullDidDetails, Web3Names } from '@kiltprotocol/did';
+
+import { BlockchainApiConnection } from '@kiltprotocol/chain-helpers';
+
+import { Option, Struct } from '@polkadot/types';
+import { AnyNumber } from '@polkadot/types/types';
+
+import * as styles from './W3NRemove.module.css';
+
+import { Identity } from '../../utilities/identities/types';
+
+import {
+  getKeypairBySeed,
+  getKeystoreFromSeed,
+} from '../../utilities/identities/identities';
+import { paths } from '../paths';
+
+import { IdentitySlide } from '../../components/IdentitySlide/IdentitySlide';
+import {
+  PasswordField,
+  usePasswordField,
+} from '../../components/PasswordField/PasswordField';
+import { KiltCurrency } from '../../components/KiltCurrency/KiltCurrency';
+import {
+  asKiltCoins,
+  KiltAmount,
+} from '../../components/KiltAmount/KiltAmount';
+import { TxStatusModal } from '../../components/TxStatusModal/TxStatusModal';
+import { LinkBack } from '../../components/LinkBack/LinkBack';
+import { Stats } from '../../components/Stats/Stats';
+
+import { getFullDidDetails } from '../../utilities/did/did';
+
+import { getExtrinsicFee } from '../../utilities/getExtrinsicFee/getExtrinsicFee';
+import { useSwrDataOrThrow } from '../../utilities/useSwrDataOrThrow/useSwrDataOrThrow';
+
+import { useSubmitStates } from '../../utilities/useSubmitStates/useSubmitStates';
+
+async function getFee(fullDid?: FullDidDetails) {
+  if (!fullDid) {
+    return undefined;
+  }
+
+  const fakeSeed = new Uint8Array(32);
+  const keypair = getKeypairBySeed(fakeSeed);
+
+  const extrinsic = await Web3Names.getReleaseByOwnerTx();
+
+  const authorized = await fullDid.authorizeExtrinsic(
+    extrinsic,
+    await getKeystoreFromSeed(fakeSeed),
+    keypair.address,
+  );
+  const blockchain = await BlockchainApiConnection.getConnectionOrConnect();
+  const signed = await blockchain.signTx(keypair, authorized);
+  return getExtrinsicFee(signed);
```

Comment:
there is `authorized.paymentInfo(keypair)` for that

Reply:
Good to know!

Reply:
Can I use the same call on an unauthorized extrinsic to get the fee? Maybe we don't need DID authorization for this extrinsic, since we're just using it to get the fee?

Reply:
Okay, after logging the values I see that they are not exactly the same, so I assume that adding DID authorization to an extrinsic increases the fee.

Reply:
yes, did authorisation wraps one extrinsic in another, leading to a bigger extrinsic. I think both fee components (didAuthorization and the wrapped extrinsic) should be additive, but I don't know what affects the didAuthorization fee (key type maybe?) and if it's constant